transport/src/main/java/io/servicefabric/transport/protocol/protostuff/ProtostuffMessageDeserializer.java

```diff
@@ -28,22 +22,34 @@ public Message deserialize(ByteBuf bb) {
 			throw new DecoderException(e.getMessage(), e);
 		}
 
+		// Deserialize data
 		Object data = deserializeData(binaryMessage);
 
+		// Convert BinaryMessage to Message
 		return new Message(binaryMessage.getQualifier(), data, binaryMessage.getCorrelationId());
 	}
 
 	public Object deserializeData(BinaryMessage message) {
+		// Handle null data
 		if (message.getData() == null) {
 			return null;
 		}
-		Class<?> clazz = typeRegistry.resolveType(message.getQualifier());
-		if (clazz != null) {
-			Schema schema = RuntimeSchema.getSchema(clazz);
+
+		// Handle binary data
+		if (message.getDataClass() == null) {
+			return message.getData();
+		}
+
+		// Handle object data
+		try {
+			Class<?> dataClass = Class.forName(message.getDataClass());
```

Comment:
I'm worried about performance of such approach. Specially in the case if class does not exists in the system which is much slower than happy case.
This code will run on each message so can be big performance issue.
We need to introduce some caching here I think.